hera_pspec/plot.py

```diff
@@ -1009,3 +1010,1272 @@ def _round_sigfig(x, up=True):
+    ax0.set_xlabel('dlys(ns)',fontsize=12)
+    ax0.semilogy()
+    
+    ax1 = fig.add_axes([0.1, 0.1, 0.4, 0.4])
+    ax1.plot(dlys, psn_real, label=r'$\sqrt{2 {Re}(P_{dd})P_{nn}}$', c='blue')
+    ax1.plot(dlys, pnn_real , label=r'$P_{nn}}$', c='orange')
+    ax1.plot(dlys, analytic_real_ori, label='analytic-original', c='g')
+    if 'diagonal' in extra_error_types:
+        ax1.plot(dlys, analytic_real_diag, label='analytic-diag', c='m')
+    if 'max' in extra_error_types:
+        ax1.plot(dlys, analytic_real_max, ls='--', label='analytic-max', c='m')
+    if 'mean' in extra_error_types:
+        ax1.plot(dlys, analytic_real_mean, ls=':', label='analytic-mean', c='m')
+    if 'min' in extra_error_types:
+        ax1.plot(dlys, analytic_real_min, ls='-.', label='analytic-min', c='m')
+    if normalized==True:
+        ax1.plot(dlys, noise[time_index], c='k', lw=0.5, label='thermal noise')
+    ax1.legend(loc='best', framealpha=0)
+    ax1.set_xticklabels('')
+    ax1.semilogy()
+
+    ax2 = fig.add_axes([0.5, 0.0, 0.4, 0.1])
+    ax2.plot(dlys, psn_imag / analytic_imag_ori , c='blue')
+    ax2.plot(dlys, pnn_imag / analytic_imag_ori,  c='orange')
+    if 'diagonal' in extra_error_types:
+        ax2.plot(dlys, analytic_imag_diag / analytic_imag_ori, c='m')
+    if 'max' in extra_error_types:
+        ax2.plot(dlys, analytic_imag_max / analytic_imag_ori, ls='--', c='m')
+    if 'mean' in extra_error_types:
+        ax2.plot(dlys, analytic_imag_mean / analytic_imag_ori, ls=':', c='m')
+    if 'min' in extra_error_types:
+        ax2.plot(dlys, analytic_imag_min / analytic_imag_ori, ls='-.', c='m')
+    if normalized==True: 
+        ax2.plot(dlys, noise[time_index]/analytic_imag_ori, c='k', lw=0.5)
+    ax2.axhline(y=1, c='g')
+    ax2.set_xlabel('dlys(ns)',fontsize=12)
+    ax2.semilogy()
+    ax2.yaxis.tick_right()
+
+    ax3 = fig.add_axes([0.5, 0.1, 0.4, 0.4])
+    ax3.plot(dlys, psn_imag, label=r'$\sqrt{2 {Re}(P_{dd})P_{nn}}$', c='blue')
+    ax3.plot(dlys, pnn_imag , label=r'$P_{nn}}$', c='orange')
+    ax3.plot(dlys, analytic_imag_ori, label='analytic-original', c='g')
+    if 'diagonal' in extra_error_types:
+        ax3.plot(dlys, analytic_imag_diag, label='analytic-diag', c='m')
+    if 'max' in extra_error_types:
+        ax3.plot(dlys, analytic_imag_max, ls='--', label='analytic-max', c='m')
+    if 'mean' in extra_error_types: 
+        ax3.plot(dlys, analytic_imag_mean, ls=':', label='analytic-mean', c='m')
+    if 'min' in extra_error_types:
+        ax3.plot(dlys, analytic_imag_min, ls='-.', label='analytic-min', c='m')
+    if normalized==True:
+        ax3.plot(dlys, noise[time_index], c='k', lw=0.5, label='thermal noise')
+    ax3.legend(loc='best', framealpha=0)
+    ax3.semilogy()
+    ax3.yaxis.tick_right()
+    ax3.set_xticklabels('')
+
+
+def plot_zscore_hist(uvp, uvp_td, key, wedge, normalized=True, inside_wedge=True, extra_error_types=[], fit_curve='Gaussian', plot_mode='normal', show_robust=True, **kwargs):
```

Comment:
We have code in `hera_stats.stats.uvp_zscore` to compute z-scores. I think probably this plotting code should go in `hera_stats`, and should use `hera_stats.stats.uvp_zscore` to compute the zscores. Let me know if you have questions on that code.

Reply:
Thanks. hera_stats.stats.uvp_zscore relies on error bars stored in uvp,stats_array. And for plot_zscore_hist, it uses the information in cov_array and a outer time-difference power spectrum. Should we include these information in stats_array?